### DISCUSSION TITLE
Container & Core Restore from Backup

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#674] [Announcements]
### Restore from Backup
Since the [2022.4](https://www.home-assistant.io/blog/2022/04/06/release-20224/#backups-for-home-assistant-core--container-installs
) release, Home Assistant Core and Container installation types has had the ability to create backups.
But there's no easy way to restore your configuration from those backups. So, I created a script 
(only works with installs created from https://tteck.github.io/Proxmox/ ) that makes it simple.

![Screenshot 2022-10-27 6 48 50 AM](https://user-images.githubusercontent.com/21342670/198265776-9d278cae-a426-4d2e-badb-1e04e76a00a7.png)

Simply run in the LXC console and follow instructions.

For Home Assistant Core
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/core-restore-from-backup.sh)"
```

For Home Assistant Container
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/container-restore-from-backup.sh)"
```
I've used the Core version for some time now (also tested Container). 
*It will install rsync if not detected.

TIP: Download your backups for safe keeping. You can use File Browser to upload them to the backup folder if needed.

**Comment by chx0003** [comment]
Nice addition.
As a suggestion for the Home Assistant Core LXC install script, you could add the installation of MySQLClient (maybe as optional). I don't use SQLlite and MySQLClient is not installed by default.
Thanks and regards!

**Reply by tteck** [reply]
I use the PostgreSQL LXC

**Reply by chx0003** [reply]
I've used the Mariadb LXC for the server, but I had to install the MySQLClient manually in the Home Assistant Core LXC, for HA to be able to connect to the MariaDB server.

**Reply by tteck** [reply]
Really, I haven't heard that before. What did you have to do `pip install mysqlclient`? 
So just changing the 
```
recorder:
  db_url: 
```
in the `configuration.yaml` didn't work? That's all I do.

**Reply by chx0003** [reply]
Before that I had to install some needed dependencies:
`sudo apt-get install python3-dev default-libmysqlclient-dev build-essential`
(it was giving me some errors before) then yes `pip3 install mysqlclient` and the config in HA.
It's not much to do manually, but it'd be nice to have it in the script if it's not much trouble.

**Reply by proddy** [reply]
I had to do the same and install the mysqlclient libs. https://github.com/tteck/Proxmox/discussions/804#discussioncomment-4383781
